File: src/kyu_6/pascals_triangle/python/pascals_triangle.py
def pascals_triangle(n):
    if n == 1:
        return [1]
    prev = pascals_triangle(n - 1)
    return prev + [1 if i == 0 or i == n -1 else prev[-i] + prev[-(i + 1)]
                   for i in range(n)]

File: src/kyu_6/pascals_triangle/js/pascalsTriangle.js
function pascalsTriangle(n) {
    if (n === 1) {
        return [1];
    }
    var prev = pascalsTriangle(n - 1), len = prev.length;
    prev.push(1);
    for (var i = len - n + 1; i < len - 1; i ++) {
        prev.push(prev[i] + prev[i + 1]);
    }
    prev.push(1);
    return prev;
}
module.exports = pascalsTriangle;